### ISSUE TITLE
Manually removing files

### ISSUE THREAD

**Issue by ohreally** [piwik/piwik#11256]
It seems that after every upgrade some files need to be removed manually to make Piwik run again.
After having used Piwik for quite some years, and another upgrade (2.16 → 3.01) that brought my Piwik installation to it's knees, I thought I'd do a big cleanup.

Before I start removing files, I'd like your advice.
This is the strategy I came up with:

Create a list of all the files in my installation:

cd &lt;piwik dir&gt;/
find . -print | sort > ~/piwik.me.files

Download the latest version from GitHub, extract it, and create a list of files in that version (same commands, different filename).

Find the difference between the 2, and delete files that no longer exist:

comm -23 ~/piwik.me.files ~/piwik.github.files

First question is 'Would this be the way to go?'.

If so, then please advice on how to use common sense with this list. E.g.

> Ignore all .png files, as we replaced .gif with .png.

> Ignore plugins/* except plugins/CorePluginsAdmin/*

> etc.

Thank you.

**Comment by ohreally** [comment]
Adding my list as an example.
[piwik_gone_files.txt](https://github.com/piwik/piwik/files/721908/piwik_gone_files.txt)

**Comment by Findus23** [comment]
Another idea: What's wrong with replacing the old piwik folder with the latest piwik version (from https://piwik.org/download/)
and just keeping the config in `config/config.ini.php`?
(Serious question, I might be missing some important files, that might get lost this way)

**Comment by ohreally** [comment]
That might be a solution, too, but I manage all my installed applications through DirectAdmin/Installatron, which is the way my hosting provider works.
And yes: I do have shell access, but I still prefer to manage everything (or as much as possible) centralized. Also, I'd like to provide my hosting provider with a workable solution for his DA users who do not have shell access (or shell knowledge).

**Comment by ohreally** [comment]
Seriously? Nobody?
There is not a single Piwik developer who could help me generate a list of files that should (or should not) be in my installation? This worries me a little...

**Comment by sgiehl** [comment]
How are you updating Piwik? The automatic process should remove all unneeded files. If not there might be insufficent access writes.
The solution suggest by @Findus23 should work. Using Git downloads to compare might not be the best solution, as Git does not include vendor libs.

**Comment by Globulopolis** [comment]
> If not there might be insufficent access writes.

I have Apache runing on Windows platform and the web-server have full access rights to filesystem, but Piwik cannot delete these files and file integrity check failed.

**Comment by ohreally** [comment]
And I run Apache on Linux (CentOS), and my web server also has full access.

Could you please tell me which file/script does the unneeded-files-check for the automated update process? That should help me get my 2 Piwik installations back up and running, that have been down for a week now.

**Comment by Findus23** [comment]
> That should help me get my 2 Piwik installations back up and running, that have been down for a week now.

What is the error message you get? Maybe it doesn't even have to do with unneeded files and can be simply solved. 
What happens if you use a fresh install (as I mentioned before)

**Comment by ohreally** [comment]
The error I get this time is "Error: ViewDataTable ID 'table' is already in use!".
The solution should be to delete plugins/CoreVisualizations/Visualizations/HtmlTable/PivotBy.php, as a websearch tells me.

For my other installation I don't remember the error message, but the solution was to delete plugins/Goals/Reports/BaseGoal.php (I think). Which I did, and I now have my dashboard back, but no hits have been recorded for a week.

For previous update errors I've had to delete plugins/CorePluginsAdmin/Tasks.php (I think), or manually empty the cache.
I don't remember the exact errors before that, but I run into trouble with almost every Piwik update, and the solution is always to manually delete files.
And since search engines always know which file(s) to delete, I know I'm not the only one running into this problem.

I haven't tried a fresh install, but I guess things would work then, because there wouldn't be any left-behind files.
